bin/rpi-security.py

```diff
@@ -190,34 +190,16 @@ def telegram_send_file(file_path):
         logger.info('Telegram file sent: %s' % file_path)
         return True
 
-def arp_ping_macs(mac_addresses, repeat=1):
+def arp_ping_macs(mac_addresses):
     """
-    Performs an ARP scan of a destination MAC address to try and determine if they are present on the network.
+    Performs an ARP scan of a destination network to try and determine if monitored devices are present on the network.
     """
-    def _arp_ping(mac_address, ip_address):
-        result = False
-        answered,unanswered = srp(Ether(dst=mac_address)/ARP(pdst=ip_address), timeout=1, verbose=False)
-        if len(answered) > 0:
-            for reply in answered:
-                if reply[1].hwsrc == mac_address:
-                    if type(result) is not list:
-                        result = []
-                    result.append(str(reply[1].psrc))
-                    result = ', '.join(result)
-        return result
-    while repeat > 0:
-        if time.time() - alarm_state['last_packet'] < 30:
-            break
-        for mac_address in mac_addresses:
-            result = _arp_ping(mac_address, config['network_address'])
-            if result:
-                logger.debug('MAC %s responded to ARP ping with address %s' % (mac_address, result))
-                break
-            else:
-                logger.debug('MAC %s did not respond to ARP ping' % mac_address)
-        if repeat > 1:
-            time.sleep(2)
-        repeat -= 1
+    logger.info("thread running")    
+    while True:
+        if time.time() - alarm_state['last_packet'] > 300:
+            logger.debug('Attempting ARP scan of subnet to wake up monitored devices')
+            answered,unanswered = srp(Ether(dst='ff:ff:ff:ff:ff:ff')/ARP(pdst=config['network_address']), timeout=1, verbose=False)
```

Comment:
OK interesting. I also suspect this is an iOS vs Andriod thing. In my testing, when trying to get reliable detection, Android was always more "awake" than iOS.

Did you make the PR because the current detection method didn't work? Or because you just thought this PR would make the code cleaner (which id does)?

About your false alarms: I am looking into using the camera for detection and then ditch the PIR sensor. It would be cleaner, cheaper and easier.